app/javascript/data/incoming_collaborators.ts

```diff
@@ -1,19 +1,12 @@
-import { cast } from "ts-safe-cast";
-
-import { request, ResponseError } from "$app/utils/request";
-
-export type IncomingCollaboratorsData = {
-  collaborators: IncomingCollaborator[];
-  collaborators_disabled_reason: string | null;
-};
+import type { CollaboratorPagesSharedProps } from "./collaborators";
 
 export type IncomingCollaborator = {
   id: string;
   seller_email: string;
   seller_name: string;
   seller_avatar_url: string;
   apply_to_all_products: boolean;
-  affiliate_percentage: number;
+  affiliate_percentage: number | null;
```

Comment:
This can be null apparently from model definition. 

I think there is a bug in production code, I was expecting `affiliate_percentage` to be `nil` since `apply_to_all_products` is "false" but it's "50". 
 
<img width="843" height="708" alt="Image" src="https://github.com/user-attachments/assets/4c8692da-a5c9-4b44-a2e8-47eeb078734a" />

Reversely, if  `apply_to_all_products` is "true". Each `ProductAffiliate` has `affiliate_percentage` set from top level.

@EmCousin Can you please confirm it this is how it's a bug or not?

Reply:
I believe this isn't a bug? Pinging again just in case this got missed